src/middleware.ts

```diff
@@ -401,37 +411,37 @@ const toThenable =
   }
 
 export const persist =
-  <S extends State>(config: StateCreator<S>, options: PersistOptions<S>) =>
-  (set: SetState<S>, get: GetState<S>, api: StoreApi<S>): S => {
-    const {
-      name,
-      getStorage = () => localStorage,
-      serialize = JSON.stringify as (state: StorageValue<S>) => string,
-      deserialize = JSON.parse as (str: string) => StorageValue<Partial<S>>,
-      blacklist,
-      whitelist,
-      partialize = (state: S) => state,
-      onRehydrateStorage,
-      version = 0,
-      migrate,
-      merge = (persistedState: any, currentState: S) => ({
+  <S extends State>(config: StateCreator<S>, _options: PersistOptions<S>) =>
```

Comment:
Not 100% sure about the use case, but we want `CustomSetState`, `CustomGetState`, `CustomStoreApi`, like some other middleware.

Reply:
I hope this helps tests/middlewareTypes.test.tsx.